presto-spark-base/src/main/java/com/facebook/presto/spark/execution/AbstractPrestoSparkQueryExecution.java

```diff
@@ -708,6 +717,144 @@ else if (writerTarget instanceof ExecutionWriterTarget.RefreshMaterializedViewHa
+            }
+
+            RddAndMore rddAndMore = new RddAndMore<T>(rdd, broadcastDependencies.build(), Optional.ofNullable(subPlan.getFragment().getPartitioningScheme().getPartitioning().getHandle()));
+            fragmentIdToRdd.put(subPlan.getFragment().getId(), rddAndMore);
+            return rddAndMore;
+        }
+    }
+
+    private boolean isBroadcastDistribution(SubPlan subPlan)
+    {
+        return subPlan.getFragment().getPartitioningScheme().getPartitioning().getHandle().equals(FIXED_BROADCAST_DISTRIBUTION);
+    }
+
+    private boolean isRootSubPlan(SubPlan subPlan)
+    {
+        return subPlan.getFragment().getId().getId() == 0;
+    }
+
+    private PrestoSparkBroadcastDependency<?> createBroadcastDependency(RddAndMore<?> childRdd)
+    {
+        PrestoSparkBroadcastDependency<?> broadcastDependency;
+        DataSize maxBroadcastMemory = getSparkBroadcastJoinMaxMemoryOverride(session);
+        if (maxBroadcastMemory == null) {
+            maxBroadcastMemory = new DataSize(min(nodeMemoryConfig.getMaxQueryBroadcastMemory().toBytes(), getQueryMaxBroadcastMemory(session).toBytes()), BYTE);
+        }
+        if (isStorageBasedBroadcastJoinEnabled(session)) {
+            validateStorageCapabilities(tempStorage);
+            TempDataOperationContext tempDataOperationContext = new TempDataOperationContext(
+                    session.getSource(),
+                    session.getQueryId().getId(),
+                    session.getClientInfo(),
+                    Optional.of(session.getClientTags()),
+                    session.getIdentity());
+
+            broadcastDependency = new PrestoSparkStorageBasedBroadcastDependency(
+                    (RddAndMore<PrestoSparkStorageHandle>) childRdd,
+                    maxBroadcastMemory,
+                    getQueryMaxTotalMemoryPerNode(session),
+                    queryCompletionDeadline,
+                    tempStorage,
+                    tempDataOperationContext,
+                    waitTimeMetrics);
+        }
+        else {
+            broadcastDependency = new PrestoSparkMemoryBasedBroadcastDependency(
+                    (RddAndMore<PrestoSparkSerializedPage>) childRdd,
+                    maxBroadcastMemory,
+                    queryCompletionDeadline,
+                    waitTimeMetrics);
+        }
+        return broadcastDependency;
+    }
+
+    protected FragmentExecutionResult executeFragment(SubPlan plan,
+            TableWriteInfo tableWriteInfo,
+            long timeout,
+            TimeUnit timeUnit,
+            Set<PrestoSparkServiceWaitTimeMetrics> waitTimeMetrics) throws SparkException, TimeoutException
+    {
+        RddAndMore rddAndMore = createRddFragmented(plan, tableWriteInfo);
+        MapOutputStatistics mapOutputStats = null;
+        List<ShuffleDependency> shuffleDependencies = rddAndMore.getShuffleDependencies();
+        // For PoS, we don't expect more than 1 shuffle dependency.
```

Comment:
why?

Reply:
We create shuffledRDD which has just one dependency : https://github.com/apache/spark/blob/f01a8db4bcf09c4975029e85722053ff82f8a355/core/src/main/scala/org/apache/spark/rdd/ShuffledRDD.scala#L87